### DISCUSSION TITLE
zig cc交叉编译链接报错_Unwind_GetCFA

### DISCUSSION THREAD

**Discussion by zhang-chuanfeng** [zigcc/forum#127] [General]
这里介绍zig cc的[交叉编](https://andrewkelley.me/post/zig-cc-powerful-drop-in-replacement-gcc-clang.html)译按照教程,
![image](https://github.com/zigcc/forum/assets/59160162/36cc3d40-cd8b-4e38-8b3e-b06bd61fa169)

在ubuntu x86下交叉编译报错  看着像和链接器相关的问题
![image](https://github.com/zigcc/forum/assets/59160162/18b7b150-9480-4e2c-89dc-62c62e741c5b)

**Comment by jiacai2050** [comment]
Andrew 那个文章好老了，估计有些过时了，你去掉交叉编译能过吗？

**Reply by zhang-chuanfeng** [reply]
> Andrew 那个文章好老了，估计有些过时了，你去掉交叉编译能过吗？

主机编译也是报链接错误  zig cc 执行完后， 直接make是直接链接成功的 可能是使用gcc的链接器

**Reply by zhang-chuanfeng** [reply]
![image](https://github.com/zigcc/forum/assets/59160162/53812ee2-c2e3-4046-8f82-b8eb10792c88)
看这里指定LD不知道 zig的链接器能不能指定

**Reply by jiacai2050** [reply]
那就不清楚了， zig cc 做交叉编译问题还是不少的，能用的情况感觉有限。
https://github.com/jiacai2050/zigcc

我之前还写了这么一个辅助的脚本，但真正用起来也还是问题频出。

**Comment by zhang-chuanfeng** [comment]
export CC="zig cc -lunwind"
make CC="$CC"

估计是zig cc里面没有实现需要动态链接

**Reply by jiacai2050** [reply]
是说加上 unwind 的链接，编译就可以了嘛？

**Reply by zhang-chuanfeng** [reply]
> 是说加上 unwind 的链接，编译就可以了嘛？

是的
